core/src/test/java/org/bitcoinj/crypto/MnemonicCodeVectorsTest.java

```diff
@@ -0,0 +1,270 @@
+package org.bitcoinj.crypto;
+
+import org.bitcoinj.core.Utils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static org.bitcoinj.core.Utils.HEX;
+import static org.bitcoinj.crypto.MnemonicCodeTest.split;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * This is a parametrized test class to execute all {@link MnemonicCode} test vectors.
+ */
+@RunWith(value = Parameterized.class)
+public class MnemonicCodeVectorsTest {
+
+    private MnemonicCode mc;
+    private String entropy;
+    private String mnemonicCode;
+    private String seed;
+    private String passphrase;
+
+    public MnemonicCodeVectorsTest(String entropy, String mnemonicCode, String seed, String passphrase) {
+        this.entropy = entropy;
+        this.mnemonicCode = mnemonicCode;
+        this.seed = seed;
+        this.passphrase = passphrase;
+    }
+
+    @Before
+    public void setup() throws IOException {
+        mc = new MnemonicCode();
+    }
+
+    @Test
+    public void testMnemonicCode() throws Exception {
+        final List<String> mnemonicCode = mc.toMnemonic(HEX.decode(entropy));
+        final byte[] seed = MnemonicCode.toSeed(mnemonicCode, passphrase);
+        final byte[] entropy = mc.toEntropy(split(this.mnemonicCode));
```

Comment:
Hmm, can't we also use Utils.SPACE_JOINER here?

Reply:
Sorry, I'm not seeing where it is applicable beyond line 66; can you specify?

Reply:
Ah, did you mean to move the "split" method to Utils, because it is now used in two different classes? Agreed; I'll do this along with the other two comments.